### ISSUE TITLE
Clarification in Regularization loss

### ISSUE THREAD

**Issue by krishnakanthnakka** [ShuangLI59/Diversity-Regularized-Spatiotemporal-Attention#2]
https://github.com/ShuangLI59/Diversity-Regularized-Spatiotemporal-Attention/blob/2bb350df501ebcc69edbc84529ab0206fc8a4cb3/trainers.py#L102

Line 102 of trainer.py, shouldn't the loss be summed for every image? as in only dimensions, 1 and 2 should be summed in the torch.sum(reg, dim=(1,2) and later we need to take mean over batch dimension.

Also, the paper in Eq.(9) uses squared loss, while the code doesn't do that. Please let me know if i got wrong.
